test/index.test.js

```diff
@@ -11,88 +11,556 @@ const should = chai.should();
+    chai.request(server)
+      .get('/v1/stops/1')
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.have.property('data').to.be.an('object');
+        const data = res.body.data;
+        data.should.have.property('id').satisfy(Number.isInteger);
+        data.should.have.property('code').to.be.a('string');
+        data.should.have.property('name').to.be.a('string');
+        data.should.have.property('desc').to.be.a('string');
+        data.should.have.property('lat').to.be.a('string');
+        data.should.have.property('lon').to.be.a('string');
+        data.should.have.property('url').to.be.a('string');
+        data.should.have.property('location-type').to.be.oneOf([0, 1]);
+        done();
+      });
+  });
+  
+  // it('should return 400 with error payload if parameter "stopId" given with incorrect format', (done) => {
+  //   chai.request(server)
+  //     .get('/v1/stops/a')
+  //     .end((err, res) => {
+  //       res.should.have.status(400);
+  //       res.body.should.have.property('error').to.be.an('array');
+  //       const errorItem = res.body.error[0];
+  //       errorItem.should.have.property('title').to.be.a('string');
+  //       errorItem.should.have.property('detail').to.be.a('string');
+  //       errorItem.should.have.property('code').satisfy(Number.isInteger)
+  //       .and.to.be.at.least(100)
+  //       .and.to.be.at.most(600);
+  //       done();
+  //     });
+  // });
+});
+
+describe('/trips/{tripId}', () => {
+  it('should return 200 with trip if parameter "tripId" given', (done) => {
+    chai.request(server)
+      .get('/v1/trips/1')
+      .end((err, res) => {
+        res.should.have.status(200);
+        res.body.should.have.property('data').to.be.an('array');
+        if(res.body.data.length > 0){
+          const dataItem = res.body.data[0];
+          dataItem.should.have.property('id').satisfy(Number.isInteger);
+          dataItem.should.have.property('route-id').satisfy(Number.isInteger);
+          dataItem.should.have.property('service-id').satisfy(Number.isInteger);
+          dataItem.should.have.property('trip-headsign').to.be.a('string');
+          dataItem.should.have.property('trip-short-name').to.be.a('string');
+          dataItem.should.have.property('direction').to.be.oneOf([0, 1]);
+          
+          dataItem.should.have.property('shape').to.be.an('array');
+          dataItem.shape[0].should.be.a('string');
+          const shapeItemElements = dataItem.shape[0].split(',');
+          shapeItemElements[0].should.have.lengthOf.at.least(1);
+          shapeItemElements[1].should.satisfy(isNumericString);
+          shapeItemElements[2].should.satisfy(isNumericString);
+          shapeItemElements[3].should.satisfy(isIntegerString);
+        }
+        done();
+      });
+  });
+  
+  it('should return 400 with error payload if parameter "tripId" not given', (done) => {
```

Comment:
อันนี้ Description กับที่เทส คนละอย่างกันนะ?
ใน description บอกว่า tripId not given แต่ในเทสเคส given แต่ผิด format